corehq/apps/auditcare/tests/test_middleware.py

```diff
@@ -69,30 +69,57 @@ def test_process_response_without_audit_doc(self):
         assert not hasattr(self.request, "audit_doc")
 
     def test_process_response_with_audit_doc_with_user(self):
-        self.request.audit_doc = audit_doc = fake_audit_doc(user="username")
+        self.request.audit_doc = audit_doc = FakeAuditDoc(user="username")
         with configured_middleware() as ware:
             ware(self.request)
         self.assertEqual(audit_doc.status_code, 200)
         self.assertEqual(audit_doc.user, "username")
-        self.assertEqual(audit_doc.save.count, 1)
+        self.assertEqual(audit_doc.save_count, 1)
 
     def test_process_response_with_audit_doc_and_audit_user(self):
-        self.request.audit_doc = audit_doc = fake_audit_doc(user=None)
+        self.request.audit_doc = audit_doc = FakeAuditDoc(user=None)
         self.request.audit_user = "audit_user"
         with configured_middleware() as ware:
             ware(self.request)
         self.assertEqual(audit_doc.status_code, 200)
         self.assertEqual(audit_doc.user, "audit_user")
-        self.assertEqual(audit_doc.save.count, 1)
+        self.assertEqual(audit_doc.save_count, 1)
 
     def test_process_response_with_audit_doc_and_couch_user(self):
-        self.request.audit_doc = audit_doc = fake_audit_doc(user=None)
+        self.request.audit_doc = audit_doc = FakeAuditDoc(user=None)
         self.request.couch_user = Config(username="couch_user")
         with configured_middleware() as ware:
             ware(self.request)
         self.assertEqual(audit_doc.status_code, 200)
         self.assertEqual(audit_doc.user, "couch_user")
-        self.assertEqual(audit_doc.save.count, 1)
+        self.assertEqual(audit_doc.save_count, 1)
+
+    def test_should_save_audit_doc_exactly_once(self):
+        def get_response(request):
+            ware.process_view(request, view, ARGS, KWARGS)
+            return Config(status_code=200)
+        view = make_view()
+        settings = Settings(AUDIT_ALL_VIEWS=True)
+        with configured_middleware(settings, get_response) as ware:
+            ware(self.request)
+        audit_doc = self.request.audit_doc
+        self.assertEqual(audit_doc.status_code, 200)
+        self.assertEqual(audit_doc.user, "username")
+        self.assertEqual(audit_doc.save_count, 1)
+
+    def test_should_save_audit_doc_on_error(self):
+        def get_response(request):
```

Comment:
Another nitpicky style remark: rather than `get_response`, it could be named something to convey how this request handler is different from the normal request handler -- i.e. `failing_handler`. Seeing
```
with configured_middleware(settings, failing_handler)...
```
gives me more indication of what is going on with the middleware call.